src/battle_ai_util.c

```diff
@@ -1588,8 +1591,18 @@ bool32 CanTargetFaintAiWithMod(u32 battlerDef, u32 battlerAtk, s32 hpMod, s32 dm
         if (dmgMod)
             dmg *= dmgMod;
 
-        if (dmg >= hpCheck)
+        // Applies modified HP percent to AI data for consideration when running CanEndureHit
+        gAiLogicData->hpPercents[battlerAtk] = (hpCheck/gBattleMons[battlerAtk].maxHP)*100;
+
+        if (dmg >= hpCheck && !(CanEndureHit(battlerDef, battlerAtk, moves[moveIndex]) && (dmgMod <= 1)))
+        {
+            // Restores original HP percent to AI data for remaining calculations
+            gAiLogicData->hpPercents[battlerAtk] = (gBattleMons[battlerAtk].hp/gBattleMons[battlerAtk].maxHP)*100;
             return TRUE;
+        }
+
+        // Restores original HP percent to AI data for remaining calculations
+        gAiLogicData->hpPercents[battlerAtk] = (gBattleMons[battlerAtk].hp/gBattleMons[battlerAtk].maxHP)*100;
```

Comment:
Probably fine to just have the one comment that explains for what it's used for. The other stuff should follow from it. 
```suggestion
        // Applies modified HP percent to AI data for consideration when running CanEndureHit
        gAiLogicData->hpPercents[battlerAtk] = (hpCheck/gBattleMons[battlerAtk].maxHP)*100;

        if (dmg >= hpCheck && !(CanEndureHit(battlerDef, battlerAtk, moves[moveIndex]) && (dmgMod <= 1)))
        {
            gAiLogicData->hpPercents[battlerAtk] = (gBattleMons[battlerAtk].hp / gBattleMons[battlerAtk].maxHP) * 100;
            return TRUE;
        }
        gAiLogicData->hpPercents[battlerAtk] = (gBattleMons[battlerAtk].hp / gBattleMons[battlerAtk].maxHP) * 100;
```